### DISCUSSION TITLE
feature-request: implement git/dolt-like table branching

### DISCUSSION THREAD

**Discussion by uptonking** [sirixdb/sirix#655] [Ideas]
- I'm looking for a storage solution for my text editor with branching and versioning.
  - sirixdb is designed as db with version history, just a perfect match for my use case.

- but how to implement branching for a table/resource? 
  - my idea is to duplicate a resource and maintain branch in application layer.

@JohannesLichtenberger do u have any idea?

- adding branching to sirixdb seems like a killer feature.
 - there's almost no solution in the db industry like sirixdb as nosql database.
 - dolt is a good inspiration and reference for sql database.

> https://github.com/dolthub/dolt Git versions files. Dolt versions tables.

**Comment by JohannesLichtenberger** [comment]
Yes, but I'm not sure how you'd handle what changed and what's unchanged between branches. We assign dense, monotonically increasing long IDs to nodes, which never change, but as it's a simple sequence generator it's not usable to compare between branches of a resource. If you add something like this at the application level I guess you'd end up with a simple git-like tool. However, Git can't be used with very large resources (even not with Git LSF).

**Comment by JohannesLichtenberger** [comment]
BTW: maybe sirix.discourse.group is still better for discussions and we should probably move the discussion over there. What do the others think?

**Reply by mosheduminer** [reply]
Is there a point in having / using both github discussions, and that forum?

**Reply by JohannesLichtenberger** [reply]
The discourse forum is kind of a media break, I guess and harder to find, but I think with overall much more functionality...
